### PR TITLE
DOC-5268 code block background

### PR DESCRIPTION
**Pull request by jamesnocentini**
https://issues.couchbase.com/browse/DOC-5268

Steps to try different background/code colors:

- Choose a style from https://highlightjs.org/static/demo/
- Find the `css` file with the same name in the [highlight.js repo](https://github.com/highlightjs/highlight.js/tree/master/src/styles)
- Copy the `css` file content to `src/css/highlight.css`

docs-site PR https://github.com/couchbase/docs-site/pull/86
Preview https://deploy-preview-86--cb-docs-staging.netlify.com/home/index.html

### CODE REVIEW

**Comment by amarantha-k** [comment]
+1 for the light code background

**Comment by eric-schneider** [comment]
@jamiltz I haven't had a chance to try out some of the highlight color schemes. Have you collected any feedback so far? It would be nice to collect it here. Perhaps you can walk us through some options in the next team meeting to spur some feedback?

**Comment by jamesnocentini** [comment]
@eric-schneider This week @amarantha-k and I brainstormed different options and found that the GitHub color scheme (w/ our grey background) might be a good one. I've updated the PR with that color scheme.

cc @couchbase/documentation-writers

**Comment by eric-schneider** [comment]
I like it @jamiltz. Though code blocks within admonitions are hard on the eyes. I think we need to handle admonitions, code blocks, and `backtick` text all at once since they are all so wrapped up together in their design.

Also, the code language doesn't display when hovering over blocks. Is this a bug or just a glitch with the preview site?

**Comment by mojavelinux** [comment]
> Though code blocks within admonitions are hard on the eyes.

Same for code blocks inside an example block, as shown in the deploy preview (see the N1QL example).

**Comment by mojavelinux** [comment]
> the code language doesn't display when hovering over blocks. Is this a bug or just a glitch with the preview site?

You can't simply copy a highlight.js stylesheet into the site as is because it will break the styles for source blocks. It's the root selector in particular that causes the problem:

```
.hljs {
  display: block;
  overflow-x: auto;
  padding: 0.5em;
  color: #333;
  background: #f8f8f8;
}
```

You should set the background and foreground color in the doc.css stylesheet. The highlight.css sheet should then only contain the styles for the span elements.

**Comment by mojavelinux** [comment]
> Copy the css file content to src/css/highlight.css

After this step, you need to a) remove the .hljs selector and add the background and foreground color to this selector in doc.css:

```css
.doc pre:not(.highlight),
.doc pre.highlight code {
```

You should also use the `var(--weight-medium)` in place of the keyword `bold` since the monospace font you're using has multiple weights and we don't want the really strong one.

(You might want to note these modifications at the top of the stylesheet so we don't forget in the future).

**Comment by jamesnocentini** [comment]
> Also, the code language doesn't display when hovering over blocks. Is this a bug or just a glitch with the preview site?

@eric-schneider Good catch, that was a bug. It's fixed now and I've added a comment for the future in `highlight.css` (https://github.com/couchbase/docs-ui/pull/55/commits/25962e66de8593251bfb563948a59768031fe8a5)

> Same for code blocks inside an example block, as shown in the deploy preview (see the N1QL example).

@mojavelinux yes, in the team meeting it was suggested to unset the background color to help with that (https://github.com/couchbase/docs-ui/pull/55/commits/f9b8be59ae91716b76483dd27f31552af04cd1b4).

> (You might want to note these modifications at the top of the stylesheet so we don't forget in the future).

@mojavelinux Good idea, done in https://github.com/couchbase/docs-ui/pull/55/commits/49a741d00a7adaf518af88313952e1f096c693e3.

**Review by simon-dew — COMMENTED** [review]
The fill color of callouts within code blocks is different from the code background color. Fill color of the callout must be the same as the background color of the code. Also, callouts in code blocks have no circle around them. This makes them hard to spot in code. Need to have a dark circle around them. Check out the callouts in the code blocks [in this section](https://deploy-preview-86--cb-docs-staging.netlify.com/server/6.0/n1ql/n1ql-language-reference/indexing-arrays.html#examples).

File: src/css/nav.css [hunk 1]
```diff
@@ -1,5 +1,5 @@
 .nav-menu {
-  background-color: var(--color-shade);
+  background-color: transparent;
```
Comment (mojavelinux) [inline src/css/nav.css:2]: This seems unrelated to the topic at hand. Perhaps make this part of a separate PR?

Comment (mojavelinux) [reply src/css/nav.css:2]: I also don't agree it's a good idea. It takes the visual distinction away, so now it looks like it's just floating out there without a boundary.

Comment (jamesnocentini) [reply src/css/nav.css:2]: After changing the code block background to `var(--color-shade);`, which was also the left nav background color, it seemed there was too much of it. So we wanted to experiment without a background on the left nav but that isn't ideal either. @simon-dew and @eric-schneider have also expressed this feedback.

**Comment by jamesnocentini** [comment]
I'll close the PR because the general consensus is that this change doesn't improve the user experience.

I've added a comment to `highlight.css` on master https://github.com/couchbase/docs-ui/commit/10853427dc1161ad2fed111ccc4ad2e23420d5e8 to document the feedback about highlight.js themes.

**Pull request closed without merging**